Assets/Scripts/Enemy/EnemyAnimationHandler.cs

```diff
@@ -0,0 +1,49 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class EnemyAnimationHandler : MonoBehaviour
+{
+    private Enemy _enemy;
+    private StateMachine _stateMachine;
+    private Animator _animator;
+    private State _currentState;
+    private UnityEngine.AI.NavMeshAgent _navMeshAgent;
+
+    void Awake()
+    {
+        _enemy = GetComponent<Enemy>();
+        _animator = GetComponent<Animator>();
+        _navMeshAgent = GetComponent<UnityEngine.AI.NavMeshAgent>();
+    }
+
+    public void AnimateAttack()
+    {
+        // Stop the nav mesh agent so the enemy stays in place while playing attack animation.
+        _navMeshAgent.isStopped = true;
+        _animator.SetTrigger("Attack");
+    }
+
+    public void AnimateIdle()
+    {
+        // Enable the nav mesh agent and make it move at regular speed while playing idle animation.
+        _navMeshAgent.isStopped = false;

```

Comment:
Should make the enemies patrol so they're more interesting. There's an example of how to patrol in my AI version and Yudi's.